trino/trino.go

```diff
@@ -1863,16 +1867,21 @@ func (s *NullSlice3Float64) Scan(value interface{}) error {
 	return nil
 }
 
+// Layout for time and timestamp WITHOUT time zone.
+// Trino can support up to 12 digits sub second precision.
+// (Requires X-Trino-Client-Capabilities: PARAMETRIC_DATETIME)
 var timeLayouts = []string{
 	"2006-01-02",
-	"15:04:05.000",
-	"2006-01-02 15:04:05.000",
+	"15:04:05.999999999999",
+	"2006-01-02 15:04:05.999999999999",
 }
 
 // Layout for time and timestamp WITH time zone.
+// Trino can support up to 12 digits sub second precision.
```

Comment:
Can we really store picoseconds in a `time.Time`? [The docs](https://pkg.go.dev/time#Time) say:
> A Time represents an instant in time with nanosecond precision.

Reply:
So but we need to be able to parse the string containing 12 digits into Go. And yes I guess the value will get truncated because Go doesn't support better than nanos

Reply:
The driver should not have to parse it just to throw away picoseconds. If it cannot handle picoseconds, it's reasonable to expect queries to be written in a way to avoid returning them (either round or truncate explicitly) and return an actionable error message explaining this.

Reply:
We now error on precision greater than nanoseconds. So reading a timestamp as `TIMESTAMP(9)` works but greater precision fails.